### PR TITLE
For MRJobinfo endpoint, call Metrics System directly in Standalone/Inmem...

### PR DESCRIPTION
**Pull request by anwar6953**
...ory (avoid JobClient because we know it will not exist)

http://builds.cask.co/browse/CDAP-RBT245-4
http://issues.cask.co/browse/CDAP-2359

Tested on Standalone. Testing in progress on Distributed.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java [hunk 1]
```diff
@@ -217,6 +222,7 @@ public Module getDistributedModules() {
                              protected void configure() {
                                bind(SchedulerService.class).to(DistributedSchedulerService.class).in(Scopes.SINGLETON);
                                bind(Scheduler.class).to(SchedulerService.class);
+                               bind(MRJobInfoFetcher.class).to(MRJobClientWithMetricsFallback.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: Shouldn't the implementation class call `LocalMRJobInfoFetcher` and `DistributedMRJobInfoFetcher`?

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: I figured the name should be more closely related to its implementation.
So `Metrics` in local and `JobClient` in distributed.
Is that not a good idea?

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: A class name should ties to what it does, not necessarily how is it implemented, hence I think `DistributedMRJobInfoFetcher` is better in this case, unless you are going to have multiple implementations of `MRJobInfoFetcher` for distributed mode.

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: Updated.

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java [hunk 2]
```diff
@@ -217,6 +222,7 @@ public Module getDistributedModules() {
                              protected void configure() {
                                bind(SchedulerService.class).to(DistributedSchedulerService.class).in(Scopes.SINGLETON);
                                bind(Scheduler.class).to(SchedulerService.class);
+                               bind(MRJobInfoFetcher.class).to(DistributedMrJobInfoFetcher.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: MR, not Mr.

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: Still not fixed

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java [hunk 3]
```diff
@@ -217,6 +222,7 @@ public Module getDistributedModules() {
                              protected void configure() {
                                bind(SchedulerService.class).to(DistributedSchedulerService.class).in(Scopes.SINGLETON);
                                bind(Scheduler.class).to(SchedulerService.class);
+                               bind(MRJobInfoFetcher.class).to(DistributedMRJobInfoFetcher.class);
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/AppFabricServiceRuntimeModule.java]: Oh.. now is fine. Browser refresh issue.

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+/**
+ * Retrieves information about a run of a MapReduce job, using {@link MRJobClient} and
+ * then {@link LocalMRJobInfoFetcher} if necessary.
+ */
+public class DistributedMRJobInfoFetcher implements MRJobInfoFetcher {
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java]: Seems like the file name doesn't match with class name. (Mr in File name). you have to turn on the git case aware.

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java]: this is because Mac FS is case insensitive.

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java]: Ok, I did this:
`git config core.ignorecase false`
Now, only one is in my local repo, but both show up in this git diff...

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java]: Ok, fixed by removing both files from git, and adding just the one.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**